### PR TITLE
feat: add Prometheus metrics endpoint

### PR DESCRIPTION
**Pull request by jacobmarble**
This change adds HTTP endpoint /metrics. Now, I need to add a metric.

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Review by mkmik — APPROVED** [review]
nice

File: src/commands/metrics.rs [hunk 1]
```diff
@@ -1,22 +1,57 @@
...
+///     .u64_counter("a.counter")
+///     .with_description("Counts things")
+///     .init();
+/// let recorder = meter
+///     .i64_value_recorder("a.value_recorder")
+///     .with_description("Records values")
+///     .init();
+///
+/// counter.add(100, &[KeyValue::new("key", "value")]);
+/// recorder.record(100, &[KeyValue::new("key", "value")]);
```
Comment (mkmik) [inline src/commands/metrics.rs:27]: thanks for the example!

how does a recorder get exported on the prometheus exported?

Comment (mkmik) [reply src/commands/metrics.rs:27]: uh I'm confused; I see it below as a histogram; I thought recorders were events; are they mapped to histograms automatically? who decides the buckets?

**Review by mkmik — APPROVED** [review]
approving again! thanks!

File: src/commands/metrics.rs (hunk 1 above)
Comment (jacobmarble) [reply src/commands/metrics.rs:27]: I just pulled these examples from [here](https://docs.rs/opentelemetry-prometheus/0.6.0/opentelemetry_prometheus/). I would like to figure out the internal metrics in a separate PR.

**Comment by jacobmarble** [comment]
Rebased...

**Pull request merged**